test/jdk/java/lang/reflect/sealed_classes/TestSecurityManagerChecks.java

```diff
@@ -0,0 +1,260 @@
+        if ("named".equals(args[0])) {
+            runNamedModuleTest();
+        } else {
+            runUnnamedModuleTest();
+        }
+    }
+
+    private static void runNamedModuleTest() throws Throwable {
+        Path classes = compileNamedModuleTest();
+        URL testLocation = TestSecurityManagerChecks.class
+                                                    .getProtectionDomain()
+                                                    .getCodeSource()
+                                                    .getLocation();
+
+        //need to use a different ClassLoader to run the test, so that the checks are performed:
+        ClassLoader testCL = new URLClassLoader(new URL[] {testLocation}, OBJECT_CL);
+        testCL.loadClass("TestSecurityManagerChecks")
+              .getDeclaredMethod("doRunNamedModuleTest", Path.class)
+              .invoke(null, classes);
+    }
+
+    public static void doRunNamedModuleTest(Path classes) throws Throwable {
+        Configuration testConfig = ModuleLayer.boot()
+                                              .configuration()
+                                              .resolve(ModuleFinder.of(),
+                                                       ModuleFinder.of(classes),
+                                                       List.of("test"));
+        ModuleLayer testLayer = ModuleLayer.boot()
+                                           .defineModulesWithOneLoader(testConfig,
+                                                                         OBJECT_CL);
+
+        // First get hold of the target classes before we enable security
+        Class<?> sealed = testLayer.findLoader("test").loadClass("test.Base");
+
+        //try without a SecurityManager:
+        Class<?>[] subclasses = sealed.getPermittedSubclasses();
+
+        if (subclasses.length != 3) {
+            throw new AssertionError("Incorrect permitted subclasses: " +
+                                       Arrays.asList(subclasses));
+        }
+
+        System.out.println("OK - getPermittedSubclasses for " + sealed.getName() +
+                           " got result: " + Arrays.asList(subclasses));
+
+        String[] denyPackageAccess = new String[1];
+        int[] checkPackageAccessCallCount = new int[1];
+
+        //try with a SecurityManager:
+        SecurityManager sm = new SecurityManager() {
+            @Override
+            public void checkPackageAccess(String pkg) {
+                if (pkg.startsWith("test.")) {
+                    checkPackageAccessCallCount[0]++;
+                }
+                if (Objects.equals(denyPackageAccess[0], pkg)) {
+                    throw new SecurityException();
+                }
+            }
+        };
+
+        System.setSecurityManager(sm);
+
+        denyPackageAccess[0] = "test";
+
+        //should pass - does not return a class from package "test":
+        sealed.getPermittedSubclasses();
```

Comment:
Adding a check to the expected returned value would make this clear (no permitted subclasses of package `test`).